src/pages/Player.vue

```diff
@@ -65,686 +77,455 @@ if (!Array.prototype.indexOf) {
-      ],
-      isBrowserSupported: true,
-      browserErrorHandlingValue: {
-        failsafeType: "g-form",
-        failsafeUrl: "",
-        youtubeId: "",
-      },
-      journey: [],
-      hasVideoPlayed: -1, // Three possible values: -1(don't know), 0(didn't play), 1(played)
-      sessionId: 0,
-      hasPlyrLoaded: false,
-      retention: [],
-      previousPlayerTime: 0,
-      configs: {},
-      isTutorialComplete: false,
-      tutorialProgress: {},
-      isTutorialUploadRequired: false,
-      isModalOnScreen: false,
-      componentProperties: {},
-      progressBarInfo: {
-        config: {},
-        progressPercent: 0,
-        totalQuestions: 0,
+      plyrConfig: {
+        controls: [
+          "play",
+          "play-large",
+          "progress",
+          "current-time",
+          "mute",
+          "volume",
+          "fullscreen",
+        ],
+
+        ratio: "16:7",
+
+        keyboard: {
+          focused: false,
+          global: false,
+        },
+
+        invertTime: false,
       },
+      //   isDataLoaded: false, // whether data has been fetched from the server
+      videoId: "", // video Id for the Plio
+      source: "unknown", // source from where the plio was accessed - can be passed as param in the plio url
+      componentProperties: {}, // properties of the plio player
+      // TODO: dummy user ID
+      userId: 1,
+      items: [], // holds the list of all items for this plio
+      itemResponses: [], // holds the responses to each item
+      videoSource: "youtube", // source for the video
+      watchTime: 0, // keeps a count of the watch time for the plio by the user
+      currentItemIndex: null, // current item being displayed
+      markerClass: [
+        // class for the item marker displayed on top of the video slider
+        "bg-red-500",
+        "absolute",
+        "z-10",
+        "transform",
+        "translate",
+        "-translate-x-2/4",
+        "translate-y-4",
+        "py-1.5",
+        "px-1",
+        "bottom-full",
+        "pointer-events-none",
+        "rounded-md",
+      ],
+      playerHeight: 0, // height of the player - updated once the player is ready
+      lastCheckTimestamp: 0, // time in milliseconds when the last check for item pop-up took place
+      isFullscreen: false, // is the player in fullscreen
+      currentTimestamp: null, // tracks the current timestamp in the video
+      plioDBId: null, // id for this plio in the plio DB table
+      sessionDBId: null, // id for this session in the plio DB table
+      retention: [], // array to store video retention value
+      lastTimestampRetention: null, // last recorded timestamp in the retention array
     };
   },
+  watch: {
+    isFullscreen(newIsFullscreen) {
+      // track the fullscreen status
+      if (newIsFullscreen) this.player.fullscreen.enter();
+      else this.player.fullscreen.exit();
+    },
+  },
   async created() {
+    // redirect to login page if unauthenticated
```

Comment:
login was breaking for me, maybe because it's still using the old API endpoint, but that's okay for now